doc/build-helpers/testers.chapter.md

```diff
@@ -229,6 +229,12 @@ pkgs.testers.runNixOSTest ({ lib, ... }: {
 
 ## `nixosTest` {#tester-nixosTest}
 
+This function is deprecated in favor of [`runNixOSTest`](#tester-runNixOSTest). The reason why it's bad is that when you pass it a function, it is invoked via `callPackage`. This is somewhat incompatible with the module system and running cross-platform:
+  - The choice of packages is ambiguous when the VM host platform is not Linux. `nixosTest` could in theory pick either:
+    - `hostPkgs` for packages to run on the VM host
+    - or `config.node.pkgs` for packages that run in one of the VMs
+  - The return value of the passed function should have been a module, but `nixosTest` does not actually allow that.
+
```

Comment:
I think you should wrap it in a warning box to grab more attention.
Also moved each sentence to its own line.

```suggestion
:::{.warning}
This function is deprecated in favor of [`runNixOSTest`](#tester-runNixOSTest).
The reason why it's bad is that when you pass it a function, it is invoked via `callPackage`.
This is somewhat incompatible with the module system and running cross-platform:
  - The choice of packages is ambiguous when the VM host platform is not Linux.
    `nixosTest` could in theory pick either:
    - `hostPkgs` for packages to run on the VM host
    - or `config.node.pkgs` for packages that run in one of the VMs
  - The return value of the passed function should have been a module, but `nixosTest` does not actually allow that.
:::
```